itest/rust/src/builtin_tests/containers/array_test.rs

```diff
@@ -473,6 +473,28 @@ fn array_should_format_with_display() {
     assert_eq!(format!("{a}"), "[]");
 }
 
+#[itest]
+fn array_sort_custom() {
+    let mut a = array![1, 2, 3, 4];
+    let func = Callable::from_fn("sort backwards", |args: &[&Variant]| {
+        let res = i32::from_variant(args[0]) > i32::from_variant(args[1]);
+        Ok(Variant::from(res))
+    });
+    a.sort_unstable_custom(func);
+    assert_eq!(a, array![4, 3, 2, 1]);
+}
+
+#[itest]
+fn array_binary_sort_custom() {
+    let a = array![5, 4, 2, 1];
+    let func = Callable::from_fn("sort backwards", |args: &[&Variant]| {
+        let res = i32::from_variant(args[0]) > i32::from_variant(args[1]);
+        Ok(Variant::from(res))
+    });
+    assert_eq!(a.clone().binary_search_custom(&1, func.clone()), 3);
```

Comment:
Why the clone here? Maybe express the intention with a short comment before.

Reply:
`a.clone()` was a carryover from the array implementation. Didn't catch that it wasn't required anymore because `.binary_search_custom()` is immutable.

`func.clone()` is probably easier to understand. I can add a comment there if you think its worthwhile to clarify, but the compiler will get mad if its gone. Use after move and all that.